src/main/java/edu/kis/powp/jobs2d/drivers/transformators/ShiftingDriverDecorator.java

```diff
@@ -0,0 +1,31 @@
+package edu.kis.powp.jobs2d.drivers.transformators;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+public class ShiftingDriverDecorator extends Job2dDriverDecorator {
+    private final int shiftInXDirection, shiftInYDirection;
+    private int shiftedX, shiftedY;
+
+    public ShiftingDriverDecorator(Job2dDriver job2dDriver, int shiftInXDirection, int shiftInYDirection) {
+        super(job2dDriver);
+        this.shiftInXDirection = shiftInXDirection;
+        this.shiftInYDirection = shiftInYDirection;
+    }
+
+    @Override
+    public void setPosition(int x, int y) {
+        shift(x, y);
+        super.setPosition(shiftedX, shiftedY);
+    }
+
+    @Override
+    public void operateTo(int x, int y) {
+        shift(x, y);
+        super.operateTo(shiftedX, shiftedY);
+    }
```

Comment:
Still, there is a code duplication in all trasformation:
transform(x, y)
super.....(transformedX, transformedY).

In this case, you can try to apply the **template method** pattern, but I suggest the **strategy** to increase code reuse!
Transformation of coords is a separate responsibility and can be independent of drivers and their specific operations.